wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -150,92 +151,137 @@ def __init__(self, hostname, protocol="https", port=8443, k_entry="api/v1", o_en
 
     def list_template(self, namespace=None):
         """Returns list of templates"""
-        namespace = namespace if namespace else self.default_namespace
-        return [t.metadata.name for t in self.o_api.list_namespaced_template(namespace).items]
+        if namespace:
+            return [t.metadata.name for t in self.o_api.list_namespaced_template(namespace).items]
+        else:
+            return [t.metadata.name for t in self.o_api.list_template_for_all_namespaces().items]
 
-    def list_docker_image(self):
+    def list_image_stream_images(self):
         """Returns list of images (Docker registry only)"""
-        entities = []
-        entities_j = self.o_api.get('image')[1]['items']
-        for entity_j in entities_j:
-            if 'dockerImageReference' not in entity_j:
-                continue
-            _, name, image_id, _ = Image.parse_docker_image_info(entity_j['dockerImageReference'])
-            entities.append(Image(self, name, image_id))
-        return entities
+        return [item for item in self.o_api.list_image().items
+                if item.docker_image_reference is not None]
 
-    def list_deployment_config(self):
+    def list_deployment_config(self, namespace=None):
         """Returns list of deployment configs"""
-        entities = []
-        entities_j = self.o_api.get('deploymentconfig')[1]['items']
-        for entity_j in entities_j:
-            meta = entity_j['metadata']
-            entity = DeploymentConfig(self, meta['name'], meta['namespace'])
-            entities.append(entity)
-        return entities
+        if namespace:
+            dc = self.o_api.list_namespaced_deployment_config(namespace=namespace).items
+        else:
+            dc = self.o_api.list_deployment_config_for_all_namespaces().items
+        return dc
+
+    def list_service(self, namespace=None):
+        """Returns list of services."""
+        if namespace:
+            svc = self.k_api.list_namespaced_service(namespace=namespace).items
+        else:
+            svc = self.k_api.list_service_for_all_namespaces().items
+        return svc
+
+    def list_replication_controller(self, namespace=None):
+        """Returns list of replication controllers"""
+        if namespace:
+            rc = self.k_api.list_namespaced_replication_controller(namespace=namespace).items
+        else:
+            rc = self.k_api.list_replication_controller_for_all_namespaces().items
+        return rc
+
+    def list_node(self):
+        """Returns list of nodes"""
+        nodes = self.k_api.list_node().items
+        return nodes
+
+    def cluster_info(self):
+        """Returns information about the cluster - number of CPUs and memory in GB"""
+        aggregate_cpu, aggregate_mem = 0, 0
+        for node in self.list_node():
+            aggregate_cpu += int(node.status.capacity['cpu'])
+            aggregate_mem += int(round(int(node.status.capacity['memory'][:-2]) * 0.00000102400))
```

Comment:
Can we have just a small comment here -- 
`# converting KiB to GB. 1KiB  = 1.024E-6 GB`

Reply:
done